Source/Blazorise/Components/DatePicker/DatePicker.razor.cs

```diff
@@ -106,17 +110,35 @@ public override async Task SetParametersAsync( ParameterView parameters )
 
         if ( ParentValidation is not null )
         {
-            if ( parameters.TryGetValue<Expression<Func<TValue>>>( nameof( DateExpression ), out var expression ) )
-                await ParentValidation.InitializeInputExpression( expression );
+            if ( datesUsed )
+            {
+                if ( parameters.TryGetValue<Expression<Func<IReadOnlyList<TValue>>>>( nameof( DatesExpression ), out var datesExpression ) )
+                    await ParentValidation.InitializeInputExpression( datesExpression );
+
+                if ( parameters.TryGetValue<string>( nameof( Pattern ), out var pattern ) )
+                {
+                    // make sure we get the newest value
+                    var value = parameters.TryGetValue<IReadOnlyList<TValue>>( nameof( Dates ), out var inDates )
```

Comment:
Don't we already do this in the begging of the method? Also probably can remove the comment in favor of renaming the variable to `newestValue`. 
As usual, if we feel like there's a need to add a comment, then maybe the code can be made more explicit by renaming variables or using methods.

Reply:
I know you know this, but it's a good reminder sometimes... :)